Gateway/Request/HostedCheckoutBuilder.php

```diff
@@ -0,0 +1,105 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Acquired Limited Payment module (https://acquired.com/)
+ *
+ * Copyright (c) 2023 Acquired.com (https://acquired.com/)
+ * See LICENSE.txt for license details.
+ */
+
+namespace Acquired\Payments\Gateway\Request;
+
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use Magento\Payment\Gateway\Helper\SubjectReader;
+use Magento\Payment\Gateway\Request\BuilderInterface;
+use Magento\Framework\UrlInterface;
+use Acquired\Payments\Exception\Command\BuilderException;
+use Acquired\Payments\Gateway\Config\Hosted\Config as HostedConfig;
+use Magento\Store\Model\StoreManagerInterface;
+use Acquired\Payments\Api\AcquiredCustomerRepositoryInterface;
+use Acquired\Payments\Model\Api\CreateAcquiredCustomer;
+
+class HostedCheckoutBuilder implements BuilderInterface
+{
+
+    /**
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        private readonly LoggerInterface $logger,
+        private readonly StoreManagerInterface $storeManager,
+        private readonly UrlInterface $urlBuilder,
+        private readonly HostedConfig $hostedConfig,
+        private readonly CreateAcquiredCustomer $createAcquiredCustomer
+    ){
+    }
+
+    /**
+     * @param array $buildSubject
+     * @return array
+     * @throws BuilderException
+     */
+    public function build(array $buildSubject): array
+    {
+        try {
+            $payment = SubjectReader::readPayment($buildSubject)->getPayment();
+            $order = $payment->getOrder();
+            $amount = (float)SubjectReader::readAmount($buildSubject);
```

Comment:
Decimal values are already stored in the database, this is just a safeguard against someone passing through "12.40" instead 12.4000 as a value (string representation).

What would you suggest instead?
PHP has no primitives with fixed precision type, I can format it to a string with fixed precision and then convert back to float if that is preferred?

Reply:
Hmmm, it's an awkward one, especially since it seems like that's how Magento behaves, too.

We'll keep as is for now unless we see issues in testing.